Source/Mosa.DeviceSystem/Bitmap.cs

```diff
@@ -20,16 +20,14 @@ public static Image CreateImage(byte[] data)
 			var height = stream.Read32(0x16);
 			var dataSectionOffset = stream.Read32(0xA);
 
-			var image = new Image(width, height);
+			var image = new Image(width, height, bpp);
 
-			int x = 0;
-			int y = height - 1;
+			uint x = 0;
+			uint y = height - 1;
 
-			for (var i = dataSectionOffset; i < dataSectionOffset + (width * height * bpp); i += bpp)
+			for (var i = dataSectionOffset; i < dataSectionOffset + image.Pixels.Size; i += bpp)
 			{
-				var color = (bpp == 4) ? stream.Read32(i) : (int)(stream.Read24(i) | 0xFF000000); // 32 & 24 bit
```

Comment:
Isn't this required? I think Read32 will break for 24bit data streams

Reply:
What do you mean? There are added checks (bpp == 4) to use Read32() and if not use Read24() (so that it doesn't break)

Reply:
That's been removed, it was there before but the code that is there now is only a Read32